pkg/client/dtclient/cache.go

```diff
@@ -0,0 +1,65 @@
+/*
+ * @license
+ * Copyright 2023 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dtclient
+
+import "sync"
+
+type cache[T any] struct {
+	cachedItems map[string][]T
+	mutex       sync.RWMutex
+}
+
+func (s *cache[T]) isCached(id string) bool {
```

Comment:
might be a stupid question / the cache usage is super specialized on filter(, but: 

Why is there no`get(id string) (T, bool)` at all? 
That would allow me to just do the _, ok check on my own where needed - following the idioms of map

Reply:
i like the idea. will do